2nd-gen/alerts-to-discord/functions/index.js

```diff
@@ -139,3 +142,64 @@ UDID **${testerDeviceIdentifier}** for ${testerDeviceModelName}
   }
 });
 // [END v2Alerts]
+
+/**
+ * Function triggered by Firebase Performance that publishes
+ * a message to Discord whenever a performance threshold alert is fired.
+ */
+// [START v2PerformanceAlertTrigger]
+exports.postperformancealerttodiscord = onThresholdAlertPublished(
+    async (event) => {
+      // [END v2PerformanceAlertTrigger]
+      // [START v2PerformanceEventPayload]
+      // construct a helpful message to send to Discord
+      const appId = event.appId;
+      const {
+        eventName,
+        metricType,
+        eventType,
+        numSamples,
+        thresholdValue,
+        thresholdUnit,
+        conditionPercentile,
+        appVersion,
+        violationValue,
+        violationUnit,
+        investigateUri,
+      } = event.data.payload;
+      const message = `
+    ⚠️ Performance Alert for ${metricType} of ${eventType}: **${eventName}** ⚠️
+    
+    App id: ${appId}
+    Alert condition: ${thresholdValue} ${thresholdUnit}
+    Percentile (if applicable): ${conditionPercentile}
+    App version (if applicable): ${appVersion}
+    
+    Violation: ${violationValue} ${violationUnit}
```

Comment:
This seems like the most important thing - consider moving it higher up in the message

Reply:
I grouped it by two clusters, the first cluster is alert condition, and the second cluster is the alert violation. This resembles how our alert email displays the information (with a focus on the alert condition as well). I'll let a tech writer provide some input for this one :)

Reply:
To provide a consistent experience across surfaces, I recommend keeping the current flow. Also, the example isn't that long, and a violation is implied in the description; I think developers will expect that it's a part of the example and will look for it.

Reply:
sounds good, I'll keep it as is